robin_stocks/robinhood/authentication.py

```diff
@@ -193,12 +201,49 @@ def login(username=None, password=None, expiresIn=86400, scope='internal', by_sm
                                  'access_token': data['access_token'],
                                  'refresh_token': data['refresh_token'],
                                  'device_token': payload['device_token']}, f)
+        
         else:
-            raise Exception(data['detail'])
+            if 'detail' in data:
+                raise Exception(data['detail'])
+            raise Exception(f"Received an error response {data}")
     else:
         raise Exception('Error: Trouble connecting to robinhood API. Check internet connection.')
     return(data)
 
+def _validate_sherrif_id(device_token:str, workflow_id:str,mfa_code:str):
+    url = "https://api.robinhood.com/pathfinder/user_machine/"
+    payload = {
+        'device_id': device_token,
+        'flow': 'suv',
+        'input':{'workflow_id': workflow_id}
+    }
+    data = request_post(url=url, payload=payload,json=True )
+    if "id" in data:
+        inquiries_url = f"https://api.robinhood.com/pathfinder/inquiries/{data['id']}/user_view/"
+        res = request_get(inquiries_url)
+        challenge_id=res['type_context']["context"]["sheriff_challenge"]["id"]
+        challenge_url = f"https://api.robinhood.com/challenge/{challenge_id}/respond/"
+        challenge_payload = {
+            'response': mfa_code
+        }
+        challenge_response = request_post(url=challenge_url, payload=challenge_payload,json=True )
+        if challenge_response["status"] == "validated":
+            inquiries_payload = {"sequence":0,"user_input":{"status":"continue"}}
+            inquiries_response = request_post(url=inquiries_url, payload=inquiries_payload,json=True )
+            if inquiries_response["type_context"]["result"] == "workflow_status_approved":
+                return
+            else:
+                raise Exception("workflow status  not approved")    
+        else:
+            raise Exception("Challenge not validated")
+    raise Exception("Id not returned in user-machine call")
+
+def _get_sherrif_challenge(token_id:str):
```

Comment:
Is this used anywhere?

Reply:
its a new function used only in one place

Reply:
Hes referring to line 241 function, not the validation handler, of which he is correct, the function starting at line 241 is not called anywhere and then the logic calls data value without it being in the args nor subscripted to anything. Just trying to help, the rest of the script seems pretty good